### ISSUE TITLE
bug: Transcribe does not store output in designated S3 bucket

### ISSUE THREAD

**Issue by kkocel** [localstack/localstack#8089]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a transcription job is scheduled I specify `OutputBucketName` and `OutputKey` to store the transcription, when the transcription job is complete, I check if output transcription is put there. Unfortunately, it's not there.

### Expected Behavior

Output transcription should be stored on S3 at the given bucket and key.

### How are you starting LocalStack?

Custom (please describe below)

### Steps To Reproduce

Using Test Containers 

Reproduced in project:  https://github.com/kkocel/local-stack-transcribe-repro

### Environment

```markdown
- OS: macOS
- LocalStack: latest
```

**Comment by viren-nadkarni** [comment]
Hi @kkocel, unfortunately my Kotlin skills are quite lacking. Instead I tried to reproduce this via CLI but couldn't do so:

```
$ awslocal s3 mb s3://foo
make_bucket: foo

$ awslocal s3 cp tests/integration/files/en-gb.wav s3://foo
upload: tests/integration/files/en-gb.wav to s3://foo/en-gb.wav                                                                                                                                                                              

$ awslocal transcribe start-transcription-job --transcription-job-name lorem --language-code en-GB --media MediaFileUri=s3://foo/en-gb.wav 

$ awslocal transcribe get-transcription-job --transcription-job-name lorem --query TranscriptionJob.Transcript.TranscriptFileUri
"s3://foo/4194af57.json"

$ awslocal s3 cp s3://foo/4194af57.json .
download: s3://foo/4194af57.json to ./4194af57.json             

$ cat ./4194af57.json
{"jobName": "lorem", "status": "COMPLETED", "results": {"transcripts": [{"transcript": "hello my name is stephen tried i was born in london
...
```

Furthermore this passes in our [test suite](https://github.com/localstack/localstack/blob/master/tests/integration/test_transcribe.py#L91-L93)

**Comment by viren-nadkarni** [comment]
Please note that in GetTranscriptionJob, AWS returns the presigned S3 URL in `TranscriptionJob.Transcript.TranscriptFileUri`, while LocalStack returns the S3 URI.

**Comment by kkocel** [comment]
@viren-nadkarni I specify `outputBucketName` and `outputKey` in `startTranscriptionJob`. This means that output transcription should be put in the output S3 bucket. I think this functionality may be missing from LocalStack.

**Comment by viren-nadkarni** [comment]
@kkocel This is indeed missing in LocalStack. I will update the issue description :+1:

**Comment by viren-nadkarni** [comment]
@kkocel Thanks for creating the PR!

Unfortunately we'll need tests for this before we can merge this. Could you perhaps use a VM or Rosetta to write the tests?

**Comment by kkocel** [comment]
@sannya-singal I checked it with the latest docker image and I get 404 when trying to get output from specified S3. I will link  to test showing this issue

**Comment by sannya-singal** [comment]
Hey @kkocel 👋, I checked the [sample code](https://github.com/kkocel/local-stack-transcribe-repro) provided by you and tried to reproduce the issue. Seems like the [params](https://github.com/kkocel/local-stack-transcribe-repro/blob/9d533e1990c27f1eabf46846d9f4a3e4227b3156/src/test/kotlin/com/example/localstacktranscriberepro/AmazonTranscriptionRequesterTest.kt#L70) getting passed in retrieving the file from s3 are incorrect. 

The `TranscriptFileUri` is `s3://$s3bucket/outputTranscription/jobName.json`, try passing `"outputTranscription/jobName.json"` [here](https://github.com/kkocel/local-stack-transcribe-repro/blob/9d533e1990c27f1eabf46846d9f4a3e4227b3156/src/test/kotlin/com/example/localstacktranscriberepro/AmazonTranscriptionRequesterTest.kt#L70) and it should work :slightly_smiling_face:

**Comment by kkocel** [comment]
@sannya-singal The AWS setup file key should be named `outputTranscription` instead of `"jobName.json"` so the introduced feature does not conform to actual AWS behavior.

**Comment by sannya-singal** [comment]
Hi @kkocel, thanks for reporting this issue. We have rolled out the fix in #8612. Can you please pull the `latest` docker image and give it a try? Thanks!

**Comment by sannya-singal** [comment]
Hey @kkocel, I'll close the issue for now. Please feel free to open a new issue if you are still experiencing any issues.
Thanks!

**Comment by kkocel** [comment]
hi @sannya-singal , the problem still occurs. you can try running a test in the repository I provided to see that there is still S3 not fund error there.
